chain/store/store.go

```diff
@@ -409,7 +420,7 @@ func (cs *ChainStore) PutTipSet(ctx context.Context, ts *types.TipSet) error {
 // MaybeTakeHeavierTipSet evaluates the incoming tipset and locks it in our
 // internal state as our new head, if and only if it is heavier than the current
 // head and does not exceed the maximum fork length.
-func (cs *ChainStore) MaybeTakeHeavierTipSet(ctx context.Context, ts *types.TipSet) error {
+func (cs *ChainStore) MaybeTakeHeavierTipSet(ctx context.Context, newTsk types.TipSetKey) error {
```

Comment:
can we unexport this method?

Reply:
and may be call it `informNewTipSet`?

Reply:
Basically, make it clear that we're telling the system "hey, we have this new tipset", maybe take it, maybe take something entirely different!